corpusreaders/src/main/java/edu/illinois/cs/cogcomp/nlp/corpusreaders/parcReader/PARC3Reader.java

```diff
@@ -0,0 +1,460 @@
+
+                    // Update POS tags
+                    POStags.add(pos);
+                    lemmas.add(lem);
+                }
+            }
+
+            TextAnnotation ta = new TextAnnotation(
+                    super.corpusName,
+                    fileStem,
+                    text.toString(),
+                    charOffsets.toArray(new IntPair[charOffsets.size()]),
+                    tokens.toArray(new String[tokens.size()]),
+                    sentTokOffset.stream().mapToInt(i->i).toArray());
+
+            if (bPopulatePOS)
+                populatePOS(ta, POStags);
+            if (bPopulateLemma)
+                populateLemma(ta, lemmas);
+
+            populateAttribution(ta, attrRelations);
+            result.add(ta);
+        }
+
+        return result;
+    }
+
+    private void updateAttributionRelation(Map<String, AttributionRelation> relation,
+                                           String relationId, String role, int tokenId) {
+        if (!relation.containsKey(relationId)) {
+            relation.put(relationId, new AttributionRelation(relationId));
+        }
+
+        AttributionRelation sourceCueContent = relation.get(relationId);
+
+        switch (role) {
+            case ROLE_SOURCE :
+                sourceCueContent.updateSourceSpan(tokenId);
+                break;
+            case ROLE_CUE:
+                sourceCueContent.updateCueSpan(tokenId);
+                break;
+            case ROLE_CONTENT:
+                sourceCueContent.updateContentSpan(tokenId);
+                break;
+        }
+    }
+
+    private void populatePOS(TextAnnotation ta, List<String> posTags) {
+        populateTokenLabelView(ta, posTags, ViewNames.POS);
+    }
+
+    private void populateLemma(TextAnnotation ta, List<String> lemma) {
+        populateTokenLabelView(ta, lemma, ViewNames.LEMMA);
+    }
+
+    private void populateTokenLabelView(TextAnnotation ta, List<String> tags, String viewName) {
+        TokenLabelView v = new TokenLabelView(viewName, ta);
+        for (int tkid = 0; tkid < tags.size(); tkid++)
+            v.addTokenLabel(tkid, tags.get(tkid),1.0D);
+        ta.addView(viewName, v);
+    }
+
+    private void populateAttribution(TextAnnotation ta, Map<String, AttributionRelation> attrRealtions) {
+        PredicateArgumentView attrRelationView = new PredicateArgumentView(VIEW_NAME, "Gold-PARC3", ta , 1.0D);
+
+        for (Map.Entry<String, AttributionRelation> ent : attrRealtions.entrySet()) {
+            String relationId = ent.getKey();
+            AttributionRelation rel = ent.getValue();
+            List<IntPair> sourceSpans = rel.getSourceSpans();
+            List<IntPair> cueSpans = rel.getCueSpans();
+            List<IntPair> contentSpans = rel.getContentSpans();
+
+            List<String> relations = new ArrayList<>();
+
+            // Process cue first, if there are no cue for this attribution relation, skip (should never happen)
```

Comment:
please keep a tally of this type of occurrence and report it in the report method (you'll need to override some base class version of it) -- https://github.com/CogComp/cogcomp-nlp/blob/master/corpusreaders/src/main/java/edu/illinois/cs/cogcomp/nlp/corpusreaders/AnnotationReader.java#L97
-- please report number of files that could not be read, along with the number of actual annotations read.  This method should give a user all the background information they need to report performance actually -- so if evaluation will break down different parts of an annotation, please report the numbers read for the different parts.